new-explorer-frontend/src/utils/MainApi.js

```diff
@@ -0,0 +1,54 @@
+class MainApi {
+  constructor({ baseUrl, authorizationCode }) {
+    this._authorization = authorizationCode;
+    this._baseUrl = baseUrl;
+  }
+
+  _checkResponse(res) {
+    if (!res.ok) {
+      return Promise.reject(res.status);
+    }
+    return res.json();
+  }
+
+  getSavedArticles() {
+    return fetch(`${this._baseUrl}/articles`, {
+      headers: { Authorization: this._authorization },
+    }).then(this._checkResponse);
+  }
+
+  postArticle(keyword, title, text, date, source, link, image) {
+    return fetch(`${this._baseUrl}/articles`, {
+      method: 'POST',
+      headers: {
+        Authorization: this._authorization,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        keyword,
+        title,
+        text,
+        date,
+        source,
+        link,
+        image,
+      }),
+    }).then(this._checkResponse);
+  }
+
+  deleteArticle(articleId) {
+    return fetch(`${this._baseUrl}/articles/${articleId}`, {
+      method: 'DELETE',
+      headers: {
+        Authorization: this._authorization,
+      },
+    }).then(this._checkResponse);
+  }
+}
+
+const mainApi = new MainApi({
+  authorizationCode: `Bearer ${localStorage.getItem('jwt')}`,
```

Comment:
```js

  authorizationCode: `Bearer ${localStorage.getItem('jwt')}`,

```

you need to add a **fresh** token here when you make a request otherwise it won't work


new users don't have `jwt` in their `localhost`

Reply:
* I can't **unsave** the articles. See here https://disk.yandex.ru/i/6LR4IEH3WNG_eQ . But the criteria say:  **When clicking on the active "Save" icon in the card block, a request to delete the card is made. After a successful request, the card is removed from the "Saved articles" page.**